commit b56d4eb749c80e3bbea563173cedaf0644f38bdc (refs/changes/48/348/1)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2016-08-25T22:52:24+00:00 (3 years, 1 month ago)
    
    OS-5629 illumos-joyent needs to do its check_rtime part
    OS-5622 libsmartsshd needlessly links to -ldoor
    OS-5623 find_elf.pl needs to use platform perl
    OS-5624 want option for alternate check_rtime exception file

diff --git a/exception_lists/check_rtime b/exception_lists/check_rtime
index 3f1e2a7ab1..2fac02ddc9 100644
--- a/exception_lists/check_rtime
+++ b/exception_lists/check_rtime
@@ -233,4 +233,18 @@ NOSYMSORT	usr/lib/amd64/libfru\.so\.1			# C++
 # unneeded library.
 UNREF_OBJ	lib/(libdevinfo|libcfgadm)\.so\.1; .*\ of\ .*SUNW,Netra-CP2300/lib/libprtdiag_psr\.so\.1
 
-
+# These commands are still built; however, they require libraries which
+# we don't ship and these programs aren't shipped themselves. This is a
+# Joyent specific addition. These exist because we don't have an
+# exception to deal with the fact that we have -zdefs issues.
+SKIP	^usr/bin/rmmount$
+SKIP	^usr/bin/tpmadm$
+SKIP	^usr/bin/volcheck$
+SKIP	^usr/bin/volrmmount$
+SKIP	^usr/lib/rmvolmgr$
+
+# The lx brand's VDSO object is a bit special. It's construction leads
+# us to end up having a bad sort. We should root cause this further and
+# either add an explicit exception for this kind of thing or better, fix
+# the issues it's upset about.
+SKIP	^usr/lib/brand/MACH(lx)/lx_vdso\.so\.1$
diff --git a/usr/src/lib/libsmartsshd/Makefile.com b/usr/src/lib/libsmartsshd/Makefile.com
index 914aab055b..c2f8c60e2a 100644
--- a/usr/src/lib/libsmartsshd/Makefile.com
+++ b/usr/src/lib/libsmartsshd/Makefile.com
@@ -34,7 +34,7 @@ SRCS =	$(OBJECTS:%.o=$(SRCDIR)/%.c)
 
 CPPFLAGS +=	-I$(SRCDIR) -D_REENTRANT -D_FILE_OFFSET_BITS=64
 LIBS =		$(DYNLIB) $(LINTLIB)
-LDLIBS +=	-lc -ldoor
+LDLIBS +=	-lc
 
 $(LINTLIB) :=	SRCS=	$(SRCDIR)/$(LINTSRC)
 
diff --git a/usr/src/tools/scripts/find_elf.pl b/usr/src/tools/scripts/find_elf.pl
index 51fa5e15f1..341d7abdcb 100644
--- a/usr/src/tools/scripts/find_elf.pl
+++ b/usr/src/tools/scripts/find_elf.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/perl5/bin/perl -w
 #
 # CDDL HEADER START
 #
diff --git a/usr/src/tools/scripts/nightly.sh b/usr/src/tools/scripts/nightly.sh
index 842c63a295..6cdc63936c 100644
--- a/usr/src/tools/scripts/nightly.sh
+++ b/usr/src/tools/scripts/nightly.sh
@@ -1977,9 +1977,16 @@ if [[ ($build_ok = y) && (($A_FLAG = y) || ($r_FLAG = y)) ]]; then
 		else
 			rtime_sflag="-s"
 		fi
+
+		if [[ -n $CHECK_RTIME_EXLIST && -f $CHECK_RTIME_EXLIST ]]; then
+			rtime_eflag="-e $CHECK_RTIME_EXLIST"
+		else
+			rtime_eflag=""
+		fi
+		echo $rtime_eflag
 		check_rtime -i -m -v $rtime_sflag -o -w $elf_ddir \
 			-D object_list  -f object_list -E runtime.err \
-			-I runtime.attr.raw
+			-I runtime.attr.raw $rtime_eflag
 		if (( $? != 0 )); then
 			build_extras_ok=n
 		fi
